helm/cluster-cloud-director/templates/machinedeployment.yaml

```diff
@@ -1,27 +1,33 @@
+{{- range .Values.nodePools }}
+---
 apiVersion: cluster.x-k8s.io/v1beta1
 kind: MachineDeployment
 metadata:
-  name: {{ include "resource.default.name" $ }}-md0
-  namespace: {{ .Release.Namespace }}
+  name: {{ include "resource.default.name" $ }}-{{ .name }}
+  namespace: {{ $.Release.Namespace }}
   labels:
-    {{- include "labels.common" . | nindent 4 }}
+    {{- include "labels.common" $ | nindent 4 }}
 spec:
   clusterName: {{ include "resource.default.name" $ }}
-  replicas: {{ .Values.worker.replicas }}
+  replicas: {{ .replicas }}
   selector:
     matchLabels: null
   template:
+    metadata:
+      labels:
+        {{- include "labels.common" $ | nindent 8 }}
```

Comment:
Please check https://github.com/giantswarm/giantswarm/issues/22441

Reply:
Solution suggested in upcoming commit

Reply:
Fixed in https://github.com/giantswarm/cluster-cloud-director/pull/9/commits/4820835ebf94e004075c77720624cd9052cbe078 can you check @erkanerol

Reply:
I am OK with removing the version label. 
@vxav  in this current form, I guess you need to remove that label from `{{- define "labels.common" -}}`